ouroboros-consensus/docs/GitProcess.md

```diff
@@ -0,0 +1,79 @@
+# Consensus git process
+
+This document describes the git history we strive for in the Consensus Team. The
+git history is a key asset for understanding and for debugging, so we take care
+to refine it before merging each PR.
+
+## Guiding principles
+
+_Messages matter_: Commit messages summarize an atomic coherent change and its
+motivation. They help reviewers do their job, future developers understand the
+past design process, troubleshooters debug changes in behavior, downstream
+developers understand recent changes and their context, and so on. In these kind
+of scenarios, commit messages are often the easiest documentation to read first.
+
+_Small and organized commits_: The commits themselves, beyond just their
+messages, also contribute to the story in their own way. Prefer small commits
+that include only necessary changes, and choose a commit order that seems most
+natural to you. Doing so often also makes it easier to write useful commit
+messages. Rebase before merging, so the repository's full story is as linear as
```

Comment:
Have we ever considered to do a rebase and then fast-forward merge to origin/master (to achieve full linearizability)? General question, not actual comment to this PR

Reply:
I have never asked. @dcoutts Was there specific logic to this decision?

My personal experience is that merge commits can themselves be useful, even if only for their timestamps and as an automated record of the PR number and description. Especially when we're making a reasonable effort to minimize the parallelism of the underlying genuine commits.